### PR TITLE
Release/v2.1.0 (master)

### PR DESCRIPTION
**Pull request by AlbertPG**

### CODE REVIEW

**Review by xavier-campos-diaz — DISMISSED** [review]

File: app/src/main/java/org/pesmypetcare/mypetcare/services/googlecalendar/GoogleCalendarAdapter.java [hunk 1]
```diff
@@ -16,47 +18,107 @@
...
+        executorService.shutdown();
     }
 
     @Override
-    public void newSecondaryCalendar(Pet pet) throws ExecutionException, InterruptedException {
-        ServiceLocator.getInstance().getGoogleCalendarManagerClient().createSecondaryCalendar(pet.getOwner()
-                .getGoogleCalendarToken(), pet.getOwner().getUsername(), pet.getName());
+    public void newSecondaryCalendar(Pet pet) {
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+        System.out.println("CALENDAR TOKEN "+ pet.getOwner().getGoogleCalendarToken());
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/googlecalendar/GoogleCalendarAdapter.java:67]: Codacy Issue found: ['+' is not preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5650471)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/ActualWalkingFragment.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/community/TrUnbanPost.java  3
         

Complexity decreasing per file
==============================
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrObtainAllPetWashes.java  -1
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrUpdateWash.java  -2
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/user/TrExistsUsername.java  -1
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/services/medicalprofile/MedicalProfileManagerAdapter.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/services/pet/PetManagerAdapter.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/googlecalendar/GoogleCalendarAdapter.java  1
         

Clones removed
==============
+ app/src/main/java/org/pesmypetcare/mypetcare/services/vetvisits/VetVisitsAdapter.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5650471&bid=18501242)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/pet/PetManagerAdapter.java [hunk 2]
```diff
@@ -87,21 +87,37 @@ public boolean registerNewPet(User user, Pet pet) {
...
                     libraryWeight.getBodyAsMap());
-            } catch (ExecutionException | InterruptedException e) {
+            } catch (MyPetCareException e) {
                 e.printStackTrace();
             }
         });
 
+        executorService.shutdown();
+        try {
+            executorService.awaitTermination(5, TimeUnit.MINUTES);
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/pet/PetManagerAdapter.java:107]: Codacy Issue found: ['5' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5650471)

**Review by marclll — DISMISSED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/community/TrUnbanPost.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/ActualWalkingFragment.java  3
         

Complexity decreasing per file
==============================
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrObtainAllPetWashes.java  -1
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrUpdateWash.java  -2
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/user/TrExistsUsername.java  -1
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/services/medicalprofile/MedicalProfileManagerAdapter.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/services/googlecalendar/GoogleCalendarAdapter.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/pet/PetManagerAdapter.java  1
         

Clones removed
==============
+ app/src/main/java/org/pesmypetcare/mypetcare/services/vetvisits/VetVisitsAdapter.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5650471&bid=18501242)

**Review by xavier-campos-diaz — DISMISSED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/ActualWalkingFragment.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/community/TrUnbanPost.java  3
         

Complexity decreasing per file
==============================
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrUpdateWash.java  -2
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/user/TrExistsUsername.java  -1
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrObtainAllPetWashes.java  -1
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/services/pet/PetManagerAdapter.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/medicalprofile/MedicalProfileManagerAdapter.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/services/googlecalendar/GoogleCalendarAdapter.java  1
         

Clones removed
==============
+ app/src/main/java/org/pesmypetcare/mypetcare/services/vetvisits/VetVisitsAdapter.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5650471&bid=18501242)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/community/TrUnbanPost.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/ActualWalkingFragment.java  3
         

Complexity decreasing per file
==============================
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/user/TrExistsUsername.java  -1
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrUpdateWash.java  -2
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrObtainAllPetWashes.java  -1
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/services/pet/PetManagerAdapter.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/medicalprofile/MedicalProfileManagerAdapter.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/services/googlecalendar/GoogleCalendarAdapter.java  1
         

Clones removed
==============
+ app/src/main/java/org/pesmypetcare/mypetcare/services/vetvisits/VetVisitsAdapter.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5650471&bid=18501242)

**Review by xavier-campos-diaz — APPROVED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/community/TrUnbanPost.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/ActualWalkingFragment.java  3
         

Complexity decreasing per file
==============================
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/user/TrExistsUsername.java  -1
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrUpdateWash.java  -2
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrObtainAllPetWashes.java  -1
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/services/pet/PetManagerAdapter.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/medicalprofile/MedicalProfileManagerAdapter.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/services/googlecalendar/GoogleCalendarAdapter.java  1
         

Clones removed
==============
+ app/src/main/java/org/pesmypetcare/mypetcare/services/vetvisits/VetVisitsAdapter.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5650471&bid=18501242)

**Review by marclll — APPROVED** [review]

**Review by santidrj — APPROVED** [review]
:+1:

**Pull request merged**